src/ComposerRequireChecker/DependencyGuesser/GuessFromInstalledComposerPackages.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ComposerRequireChecker\DependencyGuesser;
+
+use ComposerRequireChecker\JsonLoader;
+use Generator;
+use Roave\BetterReflection\BetterReflection;
+use Roave\BetterReflection\Identifier\Identifier;
+use Roave\BetterReflection\Identifier\IdentifierType;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflection\ReflectionConstant;
+use Roave\BetterReflection\Reflection\ReflectionFunction;
+use Roave\BetterReflection\Reflector\DefaultReflector;
+use Roave\BetterReflection\SourceLocator\Type\Composer\Factory\MakeLocatorForInstalledJson;
+use Roave\BetterReflection\SourceLocator\Type\MemoizingSourceLocator;
+use Roave\BetterReflection\SourceLocator\Type\SourceLocator;
+
+use function preg_match;
+use function preg_quote;
+use function sprintf;
+use function str_replace;
+
+use const DIRECTORY_SEPARATOR;
+
+final class GuessFromInstalledComposerPackages implements Guesser
+{
+    private SourceLocator $sourceLocator;
+    private string $pathRegex;
+
+    public function __construct(string $installationPath)
+    {
+        $this->sourceLocator = new MemoizingSourceLocator((new MakeLocatorForInstalledJson())(
+            $installationPath,
+            (new BetterReflection())->astLocator()
+        ));
+
+        $cleanPath       = preg_quote(sprintf('%s/vendor', str_replace(DIRECTORY_SEPARATOR, '/', $installationPath)), '@');
+        $this->pathRegex = sprintf('@^%s/(?:composer/\.\./)?([^/]+/[^/]+)/@', $cleanPath);
```

Comment:
Mentioned by @DanielBadura in https://github.com/maglnet/ComposerRequireChecker/pull/346#issuecomment-1076155484 - hacky as heck, so we may need to check this

Reply:
I thought about this now quite some time and I didnt came up with another solution :/